samples/star-wars-api/Policies.fs

```diff
@@ -0,0 +1,26 @@
+namespace FSharp.Data.GraphQL.Samples.StarWarsApi.Authorization
+
+open FSharp.Core
+open Microsoft.AspNetCore.Authorization
+
+module Policies =
+
+    let [<Literal>] CanSetMoon = "CanSetMoon"
+
+type IsCharacterRequierment (character : string list) =
+    member val Characters = character
+    interface IAuthorizationRequirement
+
+type IsCharacterHandler () =
+
+    inherit AuthorizationHandler<IsCharacterRequierment> () // Inject services from DI
+
+        override _.HandleRequirementAsync (context, requirement) =
+            Async.StartAsTask(async {
```

Comment:
If you do want to queue some future work on multi-processor machine, `Async.StartAsTask` is good, it says "hi thread pool, here is a work for you when you are available". But if you want something to execute as fast as possible right now, then consider using `Async.StartImmediateAsTask` which is using current thread instead of thread pool.

Reply:
I've changed that in all the places. Could you have a look?